mtags/src/main/scala-2/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -192,6 +192,19 @@ trait Completions { this: MetalsGlobal =>
     new Ordering[Member] {
       val queryLower = query.toLowerCase()
       val fuzzyCache = mutable.Map.empty[Symbol, Int]
+      val workspaceMemberPriorityCache = mutable.Map.empty[Symbol, Int]
```

Comment:
~~Will this cache be properly refreshed when the frequency changes?~~ (the tests seem to indicate so) Maybe we should control and cache it in ReferenceProvider? For example every time semanticdb changes we could refresh the cache for all symbols contained in the TextDocument? So it would not cause any additional time for completions?

Reply:
@tgodzik 
Added cache in ReferenceProvider. Added cache with the following rules:
1) If symbol not cache, counting files where the symbol might be contained.
2) On onDelete call for all values in cache, checked whether they could have been contained in the file; if yes, then priority of the symbol is decreased.
3) On onChange call for all values in the cache, it is checked whether they could have been contained in the old version of file and whether they could be in the new one. The priority of the symbol is adjusted accordingly.
4) On reset call cache is cleared.
I did this for the following reasons:
- Due to the unavailability of old file versions in onChange/onDelete, we cannot know which symbols they contained. And to recalculate cache, we need to know which symbols from the cache could have been in file.
- Keeping all symbols from TextDocuments in the cache is excessive. Compared to the total number of symbols, there are few symbols with identical names, and keeping them in the cache would require many redundant checks when files change.

Reply:
Thanks! Good job!